### PR TITLE
fix(features): Make `batch_has` fall back to defaults if no entity handler is registered

### PR DESCRIPTION
**Pull request by wedamija**
Currently, any features going through `batch_has` will be False in ST and self hosted. This adds a fallback to use `.has` in those cases. This doesn't perform any batching, but that should be ok in this case, since it'll just be reading from the local defaults.

### CODE REVIEW

File: src/sentry/features/manager.py [hunk 1]
```diff
@@ -263,6 +263,34 @@ def batch_has(
...
+            project_features = filter(lambda name: name.startswith("projects:"), feature_names)
+            if projects and project_features:
+                results: MutableMapping[str, Mapping[str, bool]] = {}
+                for project in projects:
+                    proj_results = results[f"project:{project.id}"] = {}
+                    for feature_name in project_features:
+                        proj_results[feature_name] = self.has(feature_name, project, actor=actor)
+                return results
+
+            org_features = filter(lambda name: name.startswith("organizations:"), feature_names)
```
Comment (scefali) [inline src/sentry/features/manager.py:276]: @wedamija I'm surprised we don't have this fallback logic already somewhere else

Comment (wedamija) [reply src/sentry/features/manager.py:276]: Yeah, me too. But I checked with others and we couldn't see a path for there to be a fallback here.

File: src/sentry/features/manager.py [hunk 2]
```diff
@@ -263,6 +263,34 @@ def batch_has(
...
+            if organization and org_features:
+                org_results = {}
+                for feature_name in org_features:
+                    org_results[feature_name] = self.has(feature_name, organization, actor=actor)
+                return {f"organization:{organization.id}": org_results}
+
+            unscoped_features = filter(
+                lambda name: not name.startswith("organizations:")
+                and not name.startswith("projects:"),
```
Comment (scefali) [inline src/sentry/features/manager.py:285]: @wedamija what about `user:` features?

Comment (scefali) [reply src/sentry/features/manager.py:285]: I know it's not working for Flagr yet but if we ever add it this would definitely be a landmine

Comment (wedamija) [reply src/sentry/features/manager.py:285]: Tbh I don't want to handle more cases that are already unhandled - I've been pulled pretty far off course by various things around features, just need to get this fixed and move forward.

**Review by scefali — APPROVED** [review]

**Pull request merged**